From db48f21ebd8a05ad4666fde38439f8ae7e61f80f Mon Sep 17 00:00:00 2001
From: Jordan Lazare <Jordan.Lazare@amd.com>
Date: Wed, 26 Aug 2015 17:22:18 -0500
Subject: [PATCH 395/401] amd/dal: Change vupdate to occur at first line of
 vert front porch

Seems to be necessary to keep vblanks/flips in sync with
usermode linux drivers

Signed-off-by: Jordan Lazare <Jordan.Lazare@amd.com>
Reviewed-by: Aric Cyr <Aric.Cyr@amd.com>
Reviewed-by: Harry Wentland <Harry.Wentland@amd.com>
Signed-off-by: Sanjay R Mehta <sanju.mehta@amd.com>
---
 drivers/gpu/drm/amd/dal/controller/dce110/timing_generator_dce110.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/dal/controller/dce110/timing_generator_dce110.c b/drivers/gpu/drm/amd/dal/controller/dce110/timing_generator_dce110.c
index f8dcf6d..ff82171 100644
--- a/drivers/gpu/drm/amd/dal/controller/dce110/timing_generator_dce110.c
+++ b/drivers/gpu/drm/amd/dal/controller/dce110/timing_generator_dce110.c
@@ -186,7 +186,7 @@ static bool enable_crtc(struct timing_generator *tg)
 
 	value = dal_read_reg(tg->ctx,
 			tg->regs[IDX_CRTC_MASTER_UPDATE_MODE]);
-	set_reg_field_value(value, 0,
+	set_reg_field_value(value, 3,
 			CRTC_MASTER_UPDATE_MODE, MASTER_UPDATE_MODE);
 	dal_write_reg(tg->ctx,
 			tg->regs[IDX_CRTC_MASTER_UPDATE_MODE], value);
-- 
1.9.1

